tests/integration/test_repo_policy_compliance.py

```diff
@@ -0,0 +1,182 @@
+    """Create a new forked branch for testing."""
+    branch_name = f"test/{uuid4()}"
+
+    main_branch = forked_github_repository.get_branch(forked_github_repository.default_branch)
+    branch_ref = forked_github_repository.create_git_ref(
+        ref=f"refs/heads/{branch_name}", sha=main_branch.commit.sha
+    )
+    branch = forked_github_repository.get_branch(branch_name)
+
+    yield branch
+
+    branch_ref.delete()
+
+
+@pytest.fixture(scope="module")
+def branch_with_unsigned_commit(
+    forked_github_branch: Branch, forked_github_repository: Repository
+):
+    """Create branch that would fail the branch protection check.
+
+    Makes the branch the default branch for the repository and makes the latest commit unsigned.
+    """
+    # Make an unsigned commit
+    forked_github_repository.create_file(
+        "test.txt", "testing", "some content", branch=forked_github_branch.name
+    )
+
+    # Change default branch so that the commit is ignored by the check for unique commits being
+    # signed
+    forked_github_repository.edit(default_branch=forked_github_branch.name)
+
+    # forked_github_branch.edit_protection seems to be broken as of version 1.59 of PyGithub.
+    # Without passing the users_bypass_pull_request_allowances the API returns a 422 indicating
+    # that None is not a valid value for bypass pull request allowances, with it there is a 422 for
+    # forks indicating that users and teams allowances can only be set on organisation
+    # repositories.
+    post_parameters = {
+        "required_status_checks": None,
+        "enforce_admins": None,
+        "required_pull_request_reviews": {"dismiss_stale_reviews": False},
+        "restrictions": None,
+    }
+    # pylint: disable=protected-access
+    forked_github_branch._requester.requestJsonAndCheck(  # type: ignore
+        "PUT",
+        forked_github_branch.protection_url,
+        headers={"Accept": Consts.mediaTypeRequireMultipleApprovingReviews},
+        input=post_parameters,
+    )
+    # pylint: enable=protected-access
+    forked_github_branch.add_required_signatures()
+
+    yield forked_github_branch
+
+    forked_github_branch.remove_protection()
+
+
+@pytest_asyncio.fixture(scope="module")
+async def app_with_unsigned_commit_repo(
+    model: Model, app_no_runner: Application, forked_github_repository: Repository
+) -> AsyncIterator[Application]:
+    """Application with a single runner on a repo with unsigned commit.
+
+    Test should ensure it returns with the application in a good state and has
+    one runner.
+    """
+    unit = app_no_runner.units[0]
+
+    await app_no_runner.set_config(
+        {"virtual-machines": "1", "path": forked_github_repository.full_name}
+    )
+    action = await unit.run_action("reconcile-runners")
+    await action.wait()
+    await model.wait_for_idle(status=ACTIVE_STATUS_NAME)
+
+    # Wait until there is one runner.
+    await assesrt_num_of_runners(unit, 1)
```

Comment:
I think there is a typo here :D `assesrt` -> `assert`

Reply:
Good catch

Reply:
This typo is already in the main branch. I will fix this in the next PR.